lib/remoteproc/remoteproc.c

```diff
@@ -926,13 +926,15 @@ remoteproc_create_virtio(struct remoteproc *rproc,
 	unsigned int num_vrings, i;
 	struct metal_list *node;
 
-#ifdef VIRTIO_DRIVER_ONLY
-	role = (role != VIRTIO_DEV_DRIVER) ? 0xFFFFFFFFUL : role;
-#endif
+	if (role == VIRTIO_DEV_DRIVER && !VIRTIO_DRIVER_SUPPORT) {
+		metal_log(METAL_LOG_ERROR, "VirtIO role is set to Driver, but Driver support is not enabled\n");
```

Comment:
Do we really need this debug message?  It adds extra footprint for a prebuild configuration issue.

Reply:
Not sure what you mean by a prebuild configuration issue? The variable `role` is passed in by the user of this API function `remoteproc_create_virtio()` at runtime. If the library was built without `VIRTIO_DRIVER_SUPPORT` and later the user wants the created virtio to have the "driver" role, then it will fail at runtime.

There is another check that both `driver` and `device` support have been disabled (which would be a build configuration issue), that is a different check.

Either way, I can remove this message and keep the old behavior of just quietly returning NULL here.